x/distribution/keeper/delegation.go

```diff
@@ -212,3 +212,69 @@ func (k Keeper) withdrawDelegationRewards(ctx sdk.Context, val stakingtypes.Vali
 
 	return finalRewards, nil
 }
+
+func (k Keeper) WithdrawDelegationRewardsWithoutSending(ctx sdk.Context, val stakingtypes.ValidatorI, del stakingtypes.DelegationI) (sdk.Coins, error) {
+	// check existence of delegator starting info
+	if !k.HasDelegatorStartingInfo(ctx, del.GetValidatorAddr(), del.GetDelegatorAddr()) {
+		return nil, types.ErrEmptyDelegationDistInfo
+	}
```

Comment:
would be good to refactor the shared code bits with the `withdrawDelegationRewards` method